client/consensus/slots/src/lib.rs

```diff
@@ -259,7 +262,12 @@ pub trait SimpleSlotWorker<B: BlockT> {
 			return None
 		}
 
-		let claim = self.claim_slot(&slot_info.chain_head, slot, &epoch_data)?;
+		let claim = {
+			// This variable is required as it doesn't compile otherwise
+			let epoch_data = epoch_data.clone();
```

Comment:
Why? What is the error?

Reply:
I this case I am confused even though I have quite a bit of experience fighting these:
```
error: future cannot be sent between threads safely
   --> client/consensus/slots/src/lib.rs:206:69
    |
206 |       ) -> Option<SlotResult<B, <Self::Proposer as Proposer<B>>::Proof>> {
    |  ________________________________________________________________________^
207 | |         let (timestamp, slot) = (slot_info.timestamp, slot_info.slot);
208 | |         let telemetry = self.telemetry();
209 | |         let logging_target = self.logging_target();
...   |
416 | |         Some(SlotResult { block: B::new(header, body), storage_proof })
417 | |     }
    | |_____^ future created by async block is not `Send`
    |
    = help: the trait `Sync` is not implemented for `<Self as SimpleSlotWorker<B>>::EpochData`
note: future is not `Send` as this value is used across an await
   --> client/consensus/slots/src/lib.rs:265:15
    |
265 |         let claim = self.claim_slot(slot_info.chain_head.clone(), slot, epoch_data.clone()).await?;
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ first, await occurs here, with `epoch_data` maybe used later...
note: `epoch_data` is later dropped here
   --> client/consensus/slots/src/lib.rs:265:93
    |
265 |         let claim = self.claim_slot(slot_info.chain_head.clone(), slot, epoch_data.clone()).await?;
    |                                                                         ----------                ^
    |                                                                         |
    |                                                                         has type `&<Self as SimpleSlotWorker<B>>::EpochData` which is not `Send`
help: consider moving this into a `let` binding to create a shorter lived borrow
   --> client/consensus/slots/src/lib.rs:265:67
    |
265 |         let claim = self.claim_slot(slot_info.chain_head.clone(), slot, epoch_data.clone()).await?;
    |                                                                         ^^^^^^^^^^^^^^^^^^
    = note: required for the cast to the object type `dyn futures::Future<Output = std::option::Option<SlotResult<B, <<Self as SimpleSlotWorker<B>>::Proposer as Proposer<B>>::Proof>>> + std::marker::Send`

error: could not compile `sc-consensus-slots` due to previous error
```

I suspect this might be one of the areas that compiler doesn't handle well yet.